### PR TITLE
Link opencodelists

### PR DESCRIPTION
**Pull request by atamborska**
Added option to input url from OpenCodlist and filtered data frame based on it.
Processed the icd10_usage df to remove "." and "NA"s. (We can reject this change for now and rediscuss later).

### CODE REVIEW

File: inst/app/ui.R [hunk 1]
```diff
@@ -25,7 +25,13 @@ ui <- page_sidebar(
...
       "description_search",
       "Search by description:",
-      "")
+      ""),
+    textInput(
+      "codelist_url",
+      "Enter url from OpenCodelists:",
+      ""),
+    tags$p("Required URL format: https://www.opencodelists.org/codelist/CodelistID/VersionTag/",
+     style = "font-size: 9px;")
```
Comment (milanwiedemann) [inline inst/app/ui.R:34]: now that we are adding more features to the sidebar I think we should visually distinguish things a bit better, we could do this with `cards()`, here one idea that follows the suggestion I made in `server.R`

```R
    card(
      card_header("Upload from OpenCodelists"),
      textInput(
        "codelist_id",
        "Enter Codelist ID:",
        placeholder = "nhsd-primary-care-domain-refsets/bp_cod",
        NULL),
      textInput(
        "codelist_version_tag",
        "Enter Version Tag/ID:",
        placeholder = "20200812",
        NULL),
      actionButton("load_codelist", "Load codelist", class = "btn-primary")
    )
```

**Review by milanwiedemann — APPROVED** [review]
This looks great, well done! It adds the feature in a simple way that works 🎉 

I left some suggestions, but I dont think we need to add any of them before merging now. I do think we will need to address things like catching errors and seperating the search functions from the data filtering in `server.R` soon and should write them down somewhere.

File: data-raw/icd10_code_usage.R [hunk 2]
```diff
@@ -126,8 +126,10 @@ icd10_usage <- icd10_code_usage_urls |>
     start_date = as.Date(
       paste0("20", str_extract_all(start_date, "\\d+"), "-04-01")),
     end_date = as.Date(
-      paste0("20", str_extract_all(end_date, "\\d+"), "-03-31"))
-  )
+      paste0("20", str_extract_all(end_date, "\\d+"), "-03-31")),
+    icd10_code = str_replace_all(icd10_code, "\\.", "")
```
[the remark this answers is not in the document]
Comment (atamborska) [reply data-raw/icd10_code_usage.R:130]: This is explained in the embdata lab repo.

File: data-raw/icd10_code_usage.R [hunk 3]
```diff
@@ -126,8 +126,10 @@ icd10_usage <- icd10_code_usage_urls |>
...
+  )|>
+  filter(!is.na(usage))
```
[the remark this answers is not in the document]
Comment (atamborska) [reply data-raw/icd10_code_usage.R:132]: Done

File: inst/app/ui.R (hunk 1 above)
Comment (atamborska) [reply inst/app/ui.R:34]: Agreed.

File: inst/app/server.R [hunk 4]
```diff
@@ -37,6 +37,12 @@ server <- function(input, output, session) {
       data <- data %>% filter(grepl(input$description_search, description, ignore.case = TRUE))
     }
 
+    if (!is.null(input$codelist_url) && input$codelist_url != "") {
+      csv_url <- paste0(input$codelist_url, "download.csv")
+      codelist <- read.csv(csv_url, header = T)[,1]
+      data <- data %>% filter(code %in% codelist)
+    }
```
[the remark this answers is not in the document]
Comment (atamborska) [reply inst/app/server.R:44]: For later implementation

**Pull request merged**